local-tests/test_update.py

```diff
@@ -20,10 +21,16 @@
 
 def query_runtime_version(nodes):
     print('Current version:')
+    versions = []
     for i, node in enumerate(nodes):
-        sys = node.rpc('system_version').result
+        sysver = node.rpc('system_version').result
         rt = node.rpc('state_getRuntimeVersion').result['specVersion']
-        print(f'  Node {i}: system: {sys}  runtime: {rt}')
+        print(f'  Node {i}: system: {sysver}  runtime: {rt}')
+        versions.append(rt)
+    versions = set(versions)
+    if len(versions) != 1:
+        print('ERROR: nodes reported different runtime versions')
```

Comment:
this actually won't trigger any errors in the pipeline (only an error message will be displayed). Consider changing return type to set (or dict (node -> version)) and use that information for for example trigger to `sys.exit(1)` later in the script.

Reply:
It's always good to be over paranoid :) I am fine with this explanation, maybe add this "In a very unlikely ... the last query." as a comment here.

Reply:
Added